### PR TITLE
[TypeDeclaration] Skip coalesce on StrictArrayParamDimFetchRector

### LINKED ISSUE #8084: Bug in StrictArrayParamDimFetchRector
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | e.g. v[0.17.7](https://github.com/rectorphp/rector/releases/tag/0.17.7)             |

## Minimal PHP Code Causing Issue

https://getrector.com/demo/3483a27e-a61d-491d-8d11-b516ec80275c

## Expected Behaviour

should be skipped

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/8084

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉 /cc @staabm

File: phpstan.neon [hunk 1]
```diff
@@ -661,3 +661,19 @@ parameters:
...
+            message: '#Public method "Rector\\Core\\Bootstrap\\RectorConfigsResolver\:\:provide\(\)" is never used#'
+            path: src/Bootstrap/RectorConfigsResolver.php
+
+        -
+            message: '#Public method "Rector\\Core\\Console\\Style\\RectorConsoleOutputStyleFactory\:\:create\(\)" is never used#'
+            path: src/Console/Style/RectorConsoleOutputStyleFactory.php
+
+        -
+            message: '#Public method "Rector\\Core\\DependencyInjection\\RectorContainerFactory\:\:createFromBootstrapConfigs\(\)" is never used#'
+            path: src/DependencyInjection/RectorContainerFactory.php
```
Comment (samsonasik) [inline phpstan.neon:679]: @TomasVotruba this is due to new update `tomasvotruba/unused-public` to 0.1.14

```
  - Upgrading tomasvotruba/unused-public (0.1.12 => 0.1.14)
```

I excluded in `phpstan.neon` as they are used.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)

**Comment by clxmstaab** [comment]
thanks for the fast fix

**Comment by staabm** [comment]
Looking at the fix, I think we should generelly skip assigns for non-array types, e.g. https://getrector.com/demo/c8ed7989-d51a-4e39-aa3c-49b088eb17dd

**Comment by samsonasik** [comment]
@staabm @clxmstaab I created PR https://github.com/rectorphp/rector-src/pull/4591 for it.

File: phpstan.neon (hunk 1 above)
Comment (staabm) [reply phpstan.neon:679]: should be fixed by https://github.com/TomasVotruba/unused-public/pull/79

**Pull request merged**